pkg/operator/encryption/statemachine/transition.go

```diff
@@ -171,7 +171,26 @@ func getDesiredEncryptionState(oldEncryptionConfig *apiserverconfigv1.Encryption
 	}
 
 	//
-	// STEP 4: with consistent read-keys and write-keys, remove every read-key other than the write-key and one last read key.
+	// STEP 4: with consistent keys remove a GR that this operator no longer manages
```

Comment:
if we add it after the `STEP 5` then we can only reduce a list once the keys are stable and the migration was successful

Reply:
I wonder why this does not happen much further up behind the loop https://github.com/openshift/library-go/pull/950/files#diff-c4e81966536385a51d736f8985effa981b911fdbe4c32e59b1bffa64852c80ddR86.